src/main/java/gregtech/common/CommonProxy.java

```diff
@@ -381,7 +386,7 @@ public void onLoadComplete(FMLLoadCompleteEvent event) {
         if (Loader.isModLoaded(GTValues.MODID_JEI) && event.getSide() == Side.CLIENT) {
             GTJeiPlugin.setupInputHandler();
         }
-        GTRecipeInput.INSTANCES = new ObjectOpenHashSet<>();
+        GTRecipeInputCache.disableCache();
```

Comment:
Why is the cache flushed in two different places like this?

Reply:
This flushing is applied at `FMLLoadCompleteEvent`, while the other one is applied at the last step of recipe re-registration (not before FMLLoadCompleteEvent)

Reply:
Why not flush only once in `FMLLoadCompleteEvent`, to allow everything to utilize the full cache during all of the game loading process, instead of twice which may potentially reduce its effectiveness after the first flush?

Reply:
The cache is flushed only once during initial load, the condition below:
```java
if (Loader.instance().hasReachedState(LoaderState.AVAILABLE))
```
prevents cache flushing at recipe registration before initial load is complete.